website/addons/googledrive/model.py

```diff
@@ -342,109 +270,26 @@ def create_waterbutler_log(self, auth, action, metadata):
             },
         )
 
+    def fetch_access_token(self):
```

Comment:
It would be good to move these auth related functions into the GoogleDriveProvider class.

Additionally, some of the logic here: https://github.com/CenterForOpenScience/osf.io/pull/2231/files#diff-faf208470d077fc6d1d761e586899632R27
could be abstracted away. For example, the Mendeley ExternalProvider implements a ```client``` property like:
```python
@property
def client(self):
    """An API session with Mendeley"""
    if not self._client:
        self._client = self._get_client({
            'access_token': self.account.oauth_key,
            'refresh_token': self.account.refresh_token,
            'expires_at': time.mktime(self.account.expires_at.timetuple()),
            'token_type': 'bearer',
        })
    return self._client
```
I can imagine the GoogleDriveProvider might check if the token is expired before returning an instance of the GoogleDriveClient.